### PR TITLE
Updates for test case Behavior scenario 10

### PR DESCRIPTION
**Pull request by olbjo**

### CODE REVIEW

**Review by pyhys — COMMENTED** [review]
Please update documentation on `pnet_reset_ind` in `pnet_api.h` regarding valid reset modes.

File: src/pf_types.h [hunk 1]
```diff
@@ -2446,6 +2446,14 @@ typedef struct pf_port
    } adjust;
 } pf_port_t;
 
+/**
+ * PDPort nvm entry.
+ */
+typedef struct pnet_dpport_nvm
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/pf_types.h:2452]: done

File: src/pf_includes.h [hunk 2]
```diff
@@ -46,6 +46,7 @@ extern "C" {
 #include "pf_ptcp.h"
 #include "pf_scheduler.h"
 #include "pf_udp.h"
+#include "pf_pdport.h"
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/pf_includes.h:49]: done

File: src/device/pf_pdport.c [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+ * @param net              InOut: The p-net stack instance
+ * @param p_ar             In:    The AR instance.
+ * @param p_write_req      In:    The IODWrite request.
+ * @param p_bytes          In:    Input data
+ * @param p_datalength     In:    Size of the data to write.
+ * @param p_result         Out:   Detailed error information.
+ * @return  0  if operation succeeded.
+ *          -1 if an error occurred.
+ */
+static int pf_write_pdport_data_adj (
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_pdport.c:349]: done

File: src/device/pf_pdport.c [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+ * @param net              InOut: The p-net stack instance
+ * @param p_ar             In:    The AR instance.
+ * @param p_write_req      In:    The IODWrite request.
+ * @param p_bytes          In:    Input data
+ * @param p_datalength     In:    Size of the data to write.
+ * @param p_result         Out:   Detailed error information.
+ * @return  0  if operation succeeded.
+ *          -1 if an error occurred.
+ */
+static int pf_write_pdport_data_check (
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_pdport.c:287]: done

File: src/device/pf_pdport.h [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+ *   - PF_IDX_DEV_PDREAL_DATA
+ *   - PF_IDX_DEV_PDEXP_DATA
+ *   - PF_IDX_SUB_PDPORT_STATISTIC
+ *
+ * @param net              InOut: The p-net stack instance
+ * @param p_ar             In:   The AR instance.
+ * @param p_write_req      In:   The IODWrite request.
+ * @param p_req_buf        In:   The request buffer.
+ * @param data_length      In:   Size of the data to write.
+ * @param p_req_pos        InOut:Position within the request buffer.
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_pdport.h:92]: done

File: src/device/pf_pdport.c [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#include <string.h>
+#include "pf_includes.h"
+#include "pf_block_writer.h"
+#include "pf_block_reader.h"
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_pdport.c:19]: Done.
Maybe a separate commit to align other files to these rules

File: src/pf_types.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (olbjo) [reply src/pf_types.h:2452]: done

File: src/pf_includes.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (olbjo) [reply src/pf_includes.h:49]: done

**Review by pyhys — APPROVED** [review]

**Pull request merged**